crates/forge/src/partition.rs

```diff
@@ -0,0 +1,111 @@
+use serde::Serialize;
+use std::{collections::HashMap, str::FromStr};
+
+#[derive(Debug, Clone, Copy, Serialize)]
+pub struct Partition {
+    index: usize,
+    total: usize,
+}
+
+#[allow(dead_code)] // TODO: Removed in later PRs
+impl Partition {
+    #[must_use]
+    pub fn index(&self) -> usize {
+        self.index
+    }
+
+    #[must_use]
+    pub fn total(&self) -> usize {
+        self.total
+    }
+}
```

Comment:
We could allow just accessing this by making these pub.

You can disallow creating struct instances with pub fields by adding this attribute to it https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute if that's what you wanted to prevent.